src/components/GraphiQL.js

```diff
@@ -895,6 +897,25 @@ function isPromise(value) {
   return typeof value === 'object' && typeof value.then === 'function';
 }
 
+function observableToPromise(observable) {
+  if ( isPromise(observable) ) {
+    return observable;
+  }
+  if ( !isObservable(observable) ) {
+    return observable;
+  }
+  return new Promise((resolve, reject) => {
```

Comment:
I think you actually want the first value, a "takeOne" kind of logic, rather than waiting for the last. If fetcher returned some sort of unclosing socket, this could never complete.

Perhaps:

```js
if (isObservable(promiseOrObservable)) {
  return new Promise((resolve, reject) => promiseOrObservable.subscribe(resolve, reject, reject));
}
return promiseOrObservable;
```